app/util/MapToClass.scala

```diff
@@ -0,0 +1,102 @@
+package util
+
+import shapeless.labelled._
+import shapeless._
+
+object MapToClass {
+  // FROM MAP TO CASE CLASS (http://stackoverflow.com/a/31641779)
+  trait FromMap[L <: HList] {
+    def apply(m: Map[String, Any]): Option[L]
+  }
+
+  trait LowPriorityFromMap {
+    implicit def hconsFromMap1[K <: Symbol, V, T <: HList]
+    (implicit
+     witness: Witness.Aux[K],
+     typeable: Typeable[V],
+     fromMapT: Lazy[FromMap[T]]
+    ): FromMap[FieldType[K, V] :: T] = new FromMap[FieldType[K, V] :: T] {
+      def apply(m: Map[String, Any]): Option[FieldType[K, V] :: T] = for {
+        v <- m.get(witness.value.name)
```

Comment:
I know these names are used really locally but I think it might be easier to follow the code if it wasn't single characters as this scala is doing complex stuff...